lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/driver/YarnContainerManager.java

```diff
@@ -399,6 +403,10 @@ private void handleNewContainer(final Container container) {
             .setNodeId(container.getNodeId().toString())
             .setResourceMemory(container.getResource().getMemory())
             .setVirtualCores(container.getResource().getVirtualCores())
+            // send the flag of federation here, not to change any
+            // APIs, dirty fix for JIRA REEF-568
+            // until the linked YARN issue is fixed YARN-2915
+            .setRackName(Boolean.toString(yarnFederation))
```

Comment:
Please file a JIRA, maybe as a sub-task to REEF-568, to remove this hotfix. Then, mark these places with `// TODO[REEF-xxx]: Remove ...`.